runtime/permissions/lib.rs

```diff
@@ -3345,6 +3401,20 @@ impl PermissionsContainer {
     Ok(())
   }
 
+  pub fn check_net_resolved_addr_is_not_denied(
```

Comment:
workaround the fact that if you do `--allow-net=localhost`, after dns resolution we'd try to look for `127.0.0.1` and deny.

maybe a better fix would be to expand `--allow-net=localhost` to `localhost` and `127.0.0.1` on startup. we'd have to think about how to get that to avoid hurting startup perf too badly